### PR TITLE
Enable scaling fluentd-gcp resources using ScalingPolicy.

### PR DESCRIPTION
**Pull request by x13n**
See https://github.com/justinsb/scaler for more details about ScalingPolicy resource.

**What this PR does / why we need it**:
This is adding a way to override fluentd-gcp resources in a running cluster. The resources syncing for fluentd-gcp is decoupled from addon manager.

**Special notes for your reviewer**:

**Release note**:
```release-note
fluentd-gcp resources can be modified via a ScalingPolicy
```

cc @kawych @justinsb

### CODE REVIEW

File: cluster/addons/fluentd-gcp/scaler-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: fluentd-gcp-scaler
+  namespace: kube-system
+  labels:
+    k8s-app: fluentd-gcp-scaler
+    version: v0.1
```
Comment (crassirostris) [inline cluster/addons/fluentd-gcp/scaler-deployment.yaml]: Use semver for consistency?

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -1979,50 +1979,57 @@ function copy-manifests {
...
   if [[ -n "${FLUENTD_GCP_MEMORY_REQUEST:-}" ]]; then
-    if [[ -n "${request_resources}" ]]; then
-      request_resources="${request_resources},"
-    fi
-    request_resources="memory=${FLUENTD_GCP_MEMORY_REQUEST}"
+    any_overrides=true
   fi
-  if [[ -n "${request_resources}" ]]; then
-    modifying_flags="${modifying_flags} --requests=${request_resources}"
+  if ! $any_overrides; then
```
Comment (crassirostris) [inline cluster/gce/gci/configure-helper.sh]: Won't it crash if there are no overrides?

File: cluster/addons/fluentd-gcp/scaler-policy.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: scalingpolicies.scalingpolicy.kope.io
```
Comment (crassirostris) [inline cluster/addons/fluentd-gcp/scaler-policy.yaml]: `kope.io`?

**Review by crassirostris — COMMENTED** [review]
I'm little bit concerned about using the short form of the resource in the scaler (looks like a pretty common name, possible collisions) + a couple of nits. Overall LGTM

File: cluster/addons/fluentd-gcp/scaler-policy.yaml (hunk 3 above)
Comment (x13n) [reply cluster/addons/fluentd-gcp/scaler-policy.yaml]: I just reused the ScalingPolicy defined by @justinsb in https://github.com/justinsb/scaler. I hope we can ultimately migrate to autoscaling controlled by this ScalingPolicy in the long term.

File: cluster/gce/gci/configure-helper.sh (hunk 2 above)
Comment (x13n) [reply cluster/gce/gci/configure-helper.sh]: Why would it? If there are no overrides, then $any_overrides equals false.

File: cluster/addons/fluentd-gcp/scaler-deployment.yaml (hunk 1 above)
Comment (x13n) [reply cluster/addons/fluentd-gcp/scaler-deployment.yaml]: Done.

File: cluster/gce/gci/configure-helper.sh (hunk 2 above)
Comment (crassirostris) [reply cluster/gce/gci/configure-helper.sh]: Don't we use strict mode, which will throw an error if the variable hasn't been explicitly defined before? I remember such error that resulted in https://github.com/kubernetes/kubernetes/pull/55950

Comment (x13n) [reply cluster/gce/gci/configure-helper.sh]: But the variable is defined, see line 1985.

**Comment by x13n** [comment]
Btw, I updated configure-helper.sh to use full resource name.

File: cluster/gce/gci/configure-helper.sh (hunk 2 above)
Comment (crassirostris) [reply cluster/gce/gci/configure-helper.sh]: Missed it, then everything's fine, thanks :)

**Review by crassirostris — APPROVED** [review]
/lgtm

**Comment by x13n** [comment]
/assign @gmarek 
Marek, can you take a look?

**Comment by mikedanese** [comment]
Hey, this is an alpha API. We can't run this in non-alpha clusters. Furthermore this API went through no review.

**Pull request merged**